packages/create-vite/index.js

```diff
@@ -254,14 +254,17 @@ async function init() {
 
   write('package.json', JSON.stringify(pkg, null, 2))
 
-  const pkgManager = /yarn/.test(process.env.npm_execpath) ? 'yarn' : 'npm'
+  const pkgManager =
+    pmFromUserAgent(process.env.npm_config_user_agent)?.name || 'npm'
```

Comment:
Looks like we can't use `?.` in node 12 here https://github.com/vitejs/vite/pull/4220/checks?check_run_id=3066644038#step:10:235, @ryanmoyo would you change this so the CI passes? Thanks!

Reply:
Of course, I will revise it when I get home.

Reply:
Thanks, we can use it in the rest of the codebase because we are transpiling it. At one point, we may want to do the same for create-vite.